### PR TITLE
feat(#282): validate fixed-sats order amount against the node's min/max before submitting

### LINKED ISSUE #282: Validate order amount against the node's min/max sats limits before submitting
The create-order form only checks that amounts are positive and min < max
(`_checkValid` in `lib/features/order/screens/add_order_screen.dart`). It never
compares the amount against the Mostro node's `min_order_amount` /
`max_order_amount`, even though the app already parses both from the node info
event (they are shown in the About screen). An out-of-range order is only
rejected by the daemon after submission.

## Expected behavior (v1 parity)

The mobile app validates before submitting (`_validateSatsRange` in
`add_order_screen.dart`): it converts the fexchange
rate, compares against the node limits, andalized
message like:

> The amount is too low. This Mostro node anSats - maxSats sats).

with a sats-only fallback when the fiat ranth
bounds of a range order validated.

## Notes

- For fixed-sats orders the check is direct
- For market-price orders appv2 currently h*
  (see comment in `add_order_screen.dart` a this
  needs porting an exchange-rate provider (s issue
  to fixed-sats first.
- Strings must be localized in the 5 `.arb`

### PR DESCRIPTION
**Pull request by codaMW**
## Problem
The create-order form only checked that a fixed-sats amount was positive (`_checkValid` in `add_order_screen.dart`). It never compared the amount against the Mostro node's `min_order_amount` / `max_order_amount`, even though the app already parses both into `MostroInstance` (shown on the About screen). An out-of-range fixed-sats order was only rejected by the daemon *after* submission (`OutOfRangeSatsAmount`).

## Fix
Read the node's min/max from `mostroNodeProvider` and, for fixed-sats orders, disable submit and show an inline message naming the accepted range when the amount is out of bounds:

> Amount must be between {min} and {max} sats for this Mostro node

- `_satsOutOfRange(...)` returns the node's range when the entered amount is out of bounds, else null.
- `build` folds this into `isValid`, so the submit button disables live as the user types.
- A defence-in-depth check in `_submit` re-verifies the range so no code path submits an out-of-range fixed-sats order.
- **Fails open**: when the node advertises no limits, or the amount isn't yet a number, submission is not blocked (a valid order is never rejected just because the limits couldn't be read).

## Scope
Per the issue, this covers **fixed-sats orders only**:
- **Market-price orders** need an exchange-rate provider to convert fiat -> sats client-side before the same check can apply. That's out of scope here and deferred to a separate issue (the issue notes this).
- **Range orders** are unaffected their amount isn't a single fixed value, so the min/max sats check doesn't apply.

## Testing
`flutter analyze` clean; strings localized in all five locales. Verified on a physical device (Nokia C31):
- A fixed-sats amount below the node's minimum (e.g. 300 against a 500 floor) disables submit and shows the range message.
- An in-range amount (e.g. 5000 within 500–500000) re-enables submit.

## Separate issue noticed while verifying
The form currently lets you toggle range mode on *and* enter a fixed sats amount, which sends both a fiat range and a single `amountSats`; the daemon rejects that as "Invalid amount". This is pre-existing and independent of the min/max validation here I can file it separately.

Closes #282.

## Summary by CodeRabbit

- **New Features**
  - Added validation for fixed-sats orders based on the selected node’s minimum and maximum limits.
  - Displays the accepted satoshi range below the price section.
  - Prevents submission when the entered amount falls outside the permitted range.
  - Added translated validation messages in English, German, Spanish, French, and Italian.

### CODE REVIEW

**Comment by codaMW** [comment]
@coderabbitai full review

File: lib/features/order/screens/add_order_screen.dart [hunk 1]
```diff
@@ -152,7 +175,18 @@ class _AddOrderScreenState extends ConsumerState<AddOrderScreen> {
...
     final isMarket = ref.read(isMarketPriceProvider);
     final fixedSatsStr = ref.read(fixedSatsProvider);
-    if (_submitting || !_checkValid(selectedMethods, customMethod, isMarket, fixedSatsStr)) return;
+    // Defence in depth: the submit button is already disabled when invalid or
+    // out of the node's sats range, but re-check here so no code path submits
+    // an out-of-range fixed-sats order (#282).
+    final node = ref.read(mostroNodeProvider).valueOrNull;
+    final outOfRange = !isMarket && fixedSatsStr.isNotEmpty
+        ? _satsOutOfRange(fixedSatsStr, node?.minOrderAmount, node?.maxOrderAmount)
+        : null;
```
[the remark this answers is not in the document]
Comment (codaMW) [reply lib/features/order/screens/add_order_screen.dart]: All three addressed, thanks good catches.

BigInt: extracted the check into a pure satsOutOfNodeRange() that uses BigInt.tryParse  BigInt.from, so amounts beyond the signed 64-bit range (which NewOrderParams accepts as BigInt) are caught instead of failing open.
One-sided limits: the check now only fires when the node advertises both a min and a max, so no more false two-sided ranges like "10 to 10". When either bound is absent the daemon stays the backstop.
Range orders: added !_isRange to all three predicates a fixed-sats value in range mode no longer shows the warning, disables Submit, or blocks _submit. Also added unit tests covering the full matrix (at/below/above each bound, missing bounds, invalid/empty input, exact bounds, whitespace, and the beyond-64-bit BigInt case). Verified on device: range-off fixed-sats below the node minimum still disables submit and shows the message; range-on no longer triggers it.

**Review by Catrya — APPROVED** [review]
tACK
